[System.Data] referencesource -> corefx

EgorBo opened the pull request:
I've replaced almost all referencesource links in corefx.common.source to corefx.
API differences (gui-compare):
-  DataSet, DataRow, DataView, etc types now don't have some designer (WinForms?) specific attributes:
```
[System.ComponentModel.EditorAttribute]
[System.ComponentModel.Designer]
[System.ComponentModel.ToolboxItem]
```
XmlDataDocument used to be public in Mono but in corefx it's internal. I create a 'public' copy of it in corefx/XmlDataDocument.cs to keep the same API we had. However, this type has [Obsolete] attribute on it .

dnfclas commented:
Hi __@EgorBo__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

marek-safar commented on mcs/class/referencesource/System.Data/System/Data/Common/DbDataAdapter.cs:
```diff
@@ -21,8 +21,8 @@ namespace System.Data.Common {
 
-        internal static readonly object ParameterValueNonNullValue = 0;
-        internal static readonly object ParameterValueNullValue = 1;
+        internal static readonly object s_parameterValueNonNullValue = 0;
```
Why did you make this change?

EgorBo replied:
It was an intermediate commit to compile build, now this file is not used at all - I am going to remove the whole mcs/class/referencesource/System.Data/ folder once I finish or I can remove files with each change..

marek-safar replied:
right, so this is a revert to previous version

marek-safar commented on mcs/class/referencesource/System.Data/System/Data/Common/DBDataPermissionAttribute.cs:
```diff
@@ -90,14 +90,4 @@ public abstract class DBDataPermissionAttribute : CodeAccessSecurityAttribute {
         }
     }
-}
-
```
Should be #if !MONO instead

marek-safar commented on mcs/class/System.Data/corefx/XmlDataDocument.cs:
```diff
@@ -0,0 +1,3138 @@
+// Licensed to the .NET Foundation under one or more agreements.
```
Don't copy files, do #if MONO with your changes in our corefx fork instead

EgorBo replied:
@marek-safar ok, I thought it should be clean from changes, but makes sense.

EgorBo commented:
@marek-safar 
fixed.
I've updated external/corefx submodule (Dec 22 2016 -> Jan 28 2017)
things I had to do:
1) Add some missing references, members
2) Add DataSetSchemaImporterExtension.cs
corefx doesn't have it and it used to be inside /referencesource/__/DataSet.cs file
3) Add System.IO.Compression/corefx/Interop.cs
https://github.com/mono/corefx/commit/c656446f01db4b902b0c06b19551cc30d9dd9c6b#diff-6ce4d5e9b4023333d4408a947dd214d5L39 -- old impl Mono used to use was deleted. I saved it to the Interop.cs file

EgorBo commented:
Ok, since there is already a PR "bump corefx" I've rolled back my commits
